exonum-java-binding/core/rust/exonum-java/src/node.rs

```diff
@@ -43,57 +44,50 @@ pub fn run_node(command: Command) -> Result<(), failure::Error> {
 
 fn create_node(config: Config) -> Result<Node, failure::Error> {
     let node_config = config.run_config.node_config.clone();
-    let events_pool_capacity = &node_config.mempool.events_pool_capacity;
+    let events_pool_capacity = &node_config.private_config.mempool.events_pool_capacity;
     let channel = NodeChannel::new(events_pool_capacity);
     let blockchain = create_blockchain(&config, &channel)?;
 
-    let node_config_path = config
-        .run_config
-        .node_config_path
-        .to_str()
-        .expect("Cannot convert node_config_path to String")
-        .to_owned();
-
     Ok(Node::with_blockchain(
         blockchain,
         channel,
-        node_config,
-        Some(node_config_path),
+        node_config.into(),
+        // TODO: use DefaultConfigManager once it is available
+        None,
     ))
 }
 
 fn create_blockchain(
     config: &Config,
     channel: &NodeChannel,
 ) -> Result<BlockchainMut, failure::Error> {
-    let node_config = &config.run_config.node_config;
+    let node_config: CoreNodeConfig = config.run_config.node_config.clone().into();
     let database = create_database(config)?;
     let keypair = node_config.service_keypair();
     let api_sender = ApiSender::new(channel.api_requests.0.clone());
 
     let blockchain = Blockchain::new(database, keypair, api_sender);
 
     let supervisor_service = supervisor_service();
-    let genesis_config = GenesisConfigBuilder::with_consensus_config(node_config.consensus.clone())
+    let genesis_config = GenesisConfigBuilder::with_consensus_config(node_config.consensus)
         .with_artifact(Supervisor.artifact_id())
         .with_instance(supervisor_service)
         .build();
 
     let rust_runtime = create_rust_runtime(channel);
     let java_runtime = create_java_runtime(&config);
 
-    BlockchainBuilder::new(blockchain, genesis_config)
+    Ok(BlockchainBuilder::new(blockchain, genesis_config)
         .with_runtime(rust_runtime)
         .with_runtime(java_runtime)
-        .build()
+        .build())
 }
 
 fn create_rust_runtime(channel: &NodeChannel) -> RustRuntime {
-    let service_factories = standard_exonum_service_factories();
-    service_factories.into_iter().fold(
-        RustRuntime::new(channel.endpoints.0.clone()),
-        |runtime, factory| runtime.with_factory(factory),
-    )
+    RustRuntimeBuilder::new()
+        .with_factory(TimeServiceFactory::default())
```

Comment:
Played with this API a lot, ServiceFactory trait sucks because of lacking `impl ServiceFactory for Box<ServiceFactory>`. So `standard_exonum_service_factories` method is unusable.